### PR TITLE
Use proper case for proper nouns and acronysms

### PR DESCRIPTION
**Pull request by eksperimental**
_And this is the last PR of the series_

Use title case for proper nouns such as:
Elixir, Erlang, Mix, EEx, IEx, Dialyzer, Rebar, Make, Git, GitHub,
Markdown, Unix, Windows, etc.

Use upper case for acronysms such as: IO, WWW, HTTP, FTP

### CODE REVIEW

File: lib/mix/lib/mix/hex.ex [hunk 1]
```diff
@@ -61,7 +61,7 @@ defmodule Mix.Hex do
   end
 
   @doc """
-  Returns the url to the Hex mirror.
+  Returns the URL to the `Hex` mirror.
```
Comment (ericmj) [inline lib/mix/lib/mix/hex.ex]: Why backticks?

**Comment by ericmj** [comment]
I think you should be more conservative in your changes. For example capitalizing "elixir" in test names doesn't help anyone and only makes git blame worse.

File: lib/mix/lib/mix/hex.ex (hunk 1 above)
Comment (eksperimental) [inline lib/mix/lib/mix/hex.ex]: oops.. i removed this one, but ended up here somehow

**Comment by eksperimental** [comment]
It depends on how you look at it. For me spelling proper nouns with lower case is a spelling mistake, and a very badic one. So I dont think we should leave spelling mistakes in the code and sacrifice quality just to avoid altering blame.

**Comment by ericmj** [comment]
It doesn't improve the code. No one reading "elixir" would misunderstand
its meaning because it's not capitalized. But it does makes the tools I use
when I program do a worse job.

I'm all for these kinds of PRs if the changes are actual improvements but I
don't feel like this is an improvement.

On Thursday, 2 June 2016, Eksperimental <notifications@github.com> wrote:

> It depends on how you look at it. For me spelling proper nouns with lower
> case is a spelling mistake, and a very badic one. So I dont think we should
> leave spelling mistakes in the code and sacrifice quality just to avoid
> altering blame.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4754#issuecomment-223281927>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AATV2uFvkvDGUjYumPzlNWKRM9cLMHsOks5qHtIBgaJpZM4IsEkW>
> .
>


-- 
Eric Meadows-Jönsson

**Comment by eksperimental** [comment]
> No one reading "elixir" would misunderstand its meaning because it's not capitalized.
With the exception of Make, no one will misunderstand the meaning when you read any of the corrected nouns in the documentation or comments (ie. if Unix is spelled "unix" people will get the meaning). Still they are not properly spelled (and I don't think nobody will object about this change), and the sole purpose of the PR is to correct those.
I cannot see the difference between documentation and comments and a test name, that users may eventually see if the test fail.

File: lib/mix/test/mix/rebar_test.exs [hunk 2]
```diff
@@ -33,20 +33,20 @@ defmodule Mix.RebarTest do
...
-  test "load rebar config" do
+  test "load Rebar config" do
     path = MixTest.Case.fixture_path("rebar_dep")
     config = Mix.Rebar.load_config(path)
     assert config[:sub_dirs] == ['apps/*']
     assert config[:SCRIPT] == 'rebar.config.script'
   end
 
-  test "execute rebar.config.script on dependency directory" do
+  test "execute Rebar.config.script on dependency directory" do
```
Comment (lexmag) [inline lib/mix/test/mix/rebar_test.exs]: I think that's a finename and should be kept intact.
I assume same valid for a test above.

Comment (eksperimental) [inline lib/mix/test/mix/rebar_test.exs]: oops.. i didn't mean this change

**Comment by joshhart** [comment]
If I may weigh in… Elixir is young but maturing fast.  One day soon, mass updates like this will no longer be reasonable, will just be too painful. We may disagree on whether that day has already arrived but it’s surely certain that it gets harder with each passing day going forward.

So I think it’s now or never for these kind of commits. While I totally understand Eric’s point of view, my personal view is that it’s _probably_ worth suffering the pain now, if only because doing it later will only be worse.

**Comment by josevalim** [comment]
Since @lexmag and @whatyouhide are the ones responsible for "styling" changes in the codebase, I am leaving this decision up to them. :)

**Comment by whatyouhide** [comment]
I am torn on this change; as @ericmj mentioned, it bloats the Git history and blame (which are very important for projects this large) without adding a lot of value: yes, comments will be more polished but the *user* of Elixir (not the developer hacking *on* Elixir) will see no difference (except for the changes in documentation, those I am 👍 on).

On the other hand, yes "elixir" is incorrect, and "Elixir" is correct. No one can argue that :) @lexmag, any decisive opinions?

**Comment by josevalim** [comment]
If Elixir is correct then we should go ahead and merge it.

On Sunday, June 5, 2016, Andrea Leopardi <notifications@github.com> wrote:

> I am torn on this change; as @ericmj <https://github.com/ericmj>
> mentioned, it bloats the Git history and blame (which are very important
> for projects this large) without adding a lot of value: yes, comments will
> be more polished but the *user* of Elixir (not the developer hacking *on*
> Elixir) will see no difference (except for the changes in documentation,
> those I am 👍 on).
>
> On the other hand, yes "elixir" is incorrect, and "Elixir" is correct. No
> one can argue that :) @lexmag <https://github.com/lexmag>, any decisive
> opinions?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4754#issuecomment-223810803>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAlblxbWeUW5YFFJfrXZlDksfIkUPIzks5qIsK9gaJpZM4IsEkW>
> .
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by whatyouhide** [comment]
@josevalim "Elixir" in particular was an example; they're all correct ("Erlang", "Rebar", and so on).

**Comment by lexmag** [comment]
I'm 👍 for all these changes. I think it's okay to affect blame but have names right.

**Comment by whatyouhide** [comment]
Ok, so the majority tends to agree on fixing this. I'll carefully review all the occurrences once more and merge 👍

File: bin/elixir.bat [hunk 3]
```diff
@@ -70,24 +70,24 @@ if "%par%"=="""" (
...
-IF """"=="%par:-pz=%"     (shift) 
-IF """"=="%par:--app=%"   (shift) 
-IF """"=="%par:--remsh=%" (shift) 
+IF """"=="%par:-e=%"      (shift)
+IF """"=="%par:-r=%"      (shift)
+IF """"=="%par:-pr=%"     (shift)
+IF """"=="%par:-pa=%"     (shift)
+IF """"=="%par:-pz=%"     (shift)
+IF """"=="%par:--app=%"   (shift)
+IF """"=="%par:--remsh=%" (shift)
```
Comment (whatyouhide) [inline bin/elixir.bat]: @eksperimental may be a good change but this PR is already big as it is, can you remove these trailing whitespace changes? 😃

**Comment by whatyouhide** [comment]
@eksperimental I added just one more comment to the changes. Also, I'm not sure about changing "makefile" to "Makefile". While makefiles are usually called `Makefile`, the [GNU Make homepage](https://www.gnu.org/software/make/), the [Wikipedia article on Make](https://en.wikipedia.org/wiki/Make_(software)), and the [Wikipedia article on makefiles](https://en.wikipedia.org/wiki/Makefile) all use "makefile" when referring to makefiles. Wdyt?

**Comment by eksperimental** [comment]
Sorry , please do all necessary changes as i do not have access to a computer

**Comment by whatyouhide** [comment]
@eksperimental will you be able to do the changes in the next few days? We can wait for a while before merging :)

**Comment by eksperimental** [comment]
No wont. Im so sorry

**Comment by eksperimental** [comment]
I meant. "No, I wont be able to"

**Comment by whatyouhide** [comment]
Ok, I'll fix the things I mentioned and merge this then.

**Comment by whatyouhide** [comment]
Merged manually in 3bd976737e4a6b5c4f508b8f3b576e16d184be36.

**Comment by eksperimental** [comment]
Thanks all for the discussion. And to Andrea for fixing and merging

**Comment by ericmj** [comment]
Are we going to reject PRs that do not capitalize "elixir"? If not we will have to be making these kinds of patches forever and ever :(

**Comment by eksperimental** [comment]
IMO question should be: Are we going to reject PRs that  have spelling mistakes?

**Comment by ericmj** [comment]
If we all agreed on the importance of all spelling mistakes we wouldn't be having this conversation.

I would not want merge to PRs that have spelling mistakes where end users see them, in the docs for example. But if a word has the wrong capitalization in a comment I would rather have the fix/feature than get hung up on grammar.

**Comment by whatyouhide** [comment]
@ericmj as I said before, I think you make a good point but I only think the debate is relevant for changing all this stuff in this PR. For next contributions, we shouldn't let in any spelling mistakes, not in comments or in code or in docs IMO, not deliberately: we will for sure let some bad stuff in because we won't notice, but if we notice, we should point it out. Alternatively, we can just merge things right away even with typos or the such (hence achieving your "not getting hung up", which can be a good point), but fix the mistakes ourselves right away. Wdyt?

**Comment by josevalim** [comment]
> If we all agreed on the importance of all spelling mistakes we wouldn't be having this conversation.

Once such PRs are merged or the issues are opened, I automatically add a new rule to my brain where we need to properly case names.

I know that in a way it is annoying having to remember all of those rules but I believe we should aim to be grammatically correct. If someone adds a PR where they don't properly case Elixir, then we should ask it to be fixed. The reason why Elixir was not cased properly in the first place is because I didn't know better. :)

**Pull request closed without merging**